exampleAudioSink/src/main/java/com/twilio/exampleaudiosink/AudioSinkActivity.java

```diff
@@ -0,0 +1,608 @@
+     * tracks of participants. For example, if one participant publishes a video track with
+     * ScreenCapturer and CameraCapturer with the names "screen" and "camera" respectively then
+     * other participants can use RemoteVideoTrack#getName to determine which video track is
+     * produced from the other participant's screen or camera.
+     */
+    private static final String LOCAL_AUDIO_TRACK_NAME = "mic";
+
+    /*
+     * You must provide a Twilio Access Token to connect to the Video service
+     */
+    private static final String TWILIO_ACCESS_TOKEN = BuildConfig.TWILIO_ACCESS_TOKEN;
+    private static final String ACCESS_TOKEN_SERVER = BuildConfig.TWILIO_ACCESS_TOKEN_SERVER;
+
+    /*
+     * Access token used to connect. This field will be set either from the console generated token
+     * or the request to the token server.
+     */
+    private String accessToken;
+
+    /*
+     * A Room represents communication between a local participant and one or more participants.
+     */
+    private Room room;
+
+    /*
+     * Android application UI elements
+     */
+    private LocalAudioTrack localAudioTrack;
+
+    private FloatingActionButton connectActionFab;
+
+    private ImageView participantAvatarImageView;
+    private ImageButton toggleAudioSinkButton, togglePlayAudioButton;
+    private TextView audioSinkStatusText;
+    private AlertDialog connectDialog;
+    private AudioManager audioManager;
+
+    private WavFileHelper wavFileHelper;
+    private MediaPlayerHelper mediaPlayerHelper;
+    private AudioSink audioSink = new AudioSink() {
+        @Override
+        public void renderSample(@NonNull ByteBuffer audioSample, int encoding, int sampleRate, int channels) {
+            try {
+                wavFileHelper.writeBytesToFile(audioSample, encoding, sampleRate, channels);
+            } catch (IOException e) {
+                Log.e(TAG, String.format("A fatal error has occurred. Stacktrace %s", e.getLocalizedMessage()));
+            }
+        }
+    };
+
+    private int previousAudioMode;
+    private boolean previousMicrophoneMute;
+    private boolean disconnectedFromOnDestroy;
+    private boolean isSpeakerPhoneEnabled = true;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_audio_sink);
+
+        /*
+         * Enable changing the volume using the up/down keys during a conversation
+         */
+        setVolumeControlStream(AudioManager.STREAM_VOICE_CALL);
+
+        /*
+         * Needed for setting/abandoning audio focus during call
+         */
+        audioManager = (AudioManager) getSystemService(Context.AUDIO_SERVICE);
+        audioManager.setSpeakerphoneOn(isSpeakerPhoneEnabled);
+
+        /*
+         * Check camera and microphone permissions. Needed in Android M.
+         */
+        if (!checkPermissionForCameraAndMicrophone()) {
```

Comment:
Update the permission check method to only request mic access. The camera is not used in this example